### PR TITLE
Update useScaffoldEventHistory hook

### PR DESCRIPTION
**Pull request by rin-st**
## Description

1. Moved `readEvents` to `useCallback`, so it's clear which dependencies it has
2. Since `fromBlock` is required prop, removed if's like `if (fromBlock)`
3. Combined two useEffects with the same functionality `readEvents();` and
```
if (!deployedContractLoading) {
    readEvents();
}
```
The one without `if (!deployedContractLoading)` was buggy - on load it threw errors to the console 
4. Changed default state of events to `[]`, so we don't guess is it array or undefined

## Additional Information

- [x] I have read the [contributing docs](/scaffold-eth/scaffold-eth-2/blob/main/CONTRIBUTING.md) (if this is your first contribution)
- [x] This is not a duplicate of any [existing pull request](https://github.com/scaffold-eth/scaffold-eth-2/pulls)

### CODE REVIEW

File: packages/nextjs/hooks/scaffold-eth/useScaffoldEventHistory.ts [hunk 1]
```diff
@@ -45,7 +45,7 @@ export const useScaffoldEventHistory = <
   watch,
   enabled = true,
 }: UseScaffoldEventHistoryConfig<TContractName, TEventName, TBlockData, TTransactionData, TReceiptData>) => {
-  const [events, setEvents] = useState<any[]>();
+  const [events, setEvents] = useState<any[]>([]);
   const [isLoading, setIsLoading] = useState(false);
```
Comment (damianmarti) [inline packages/nextjs/hooks/scaffold-eth/useScaffoldEventHistory.ts:49]: should we change the default value to true?

**Review by damianmarti — APPROVED** [review]
Works well and the code looks good to me, I only left a small comment.

Thanks!!

File: packages/nextjs/hooks/scaffold-eth/useScaffoldEventHistory.ts [hunk 2]
```diff
@@ -56,95 +56,89 @@ export const useScaffoldEventHistory = <
...
+      JSON.stringify(filters, replacer),
+      fromBlockUpdated,
+      publicClient,
+      receiptData,
+      transactionData,
+    ],
+  );
 
   useEffect(() => {
     if (!deployedContractLoading) {
```
Comment (Pabl0cks) [inline packages/nextjs/hooks/scaffold-eth/useScaffoldEventHistory.ts:138]: With the updated version, when there is an error, it keeps spamming the `readEvents` function, and we keep getting more and more errors in loop.

Should we add `&& !error` to this if condition?

Screenshot of the error spam:
image

**Review by Pabl0cks — COMMENTED** [review]
Good job @rin-st!!

Tested it nicely and only found a problem when you pass `enabled: false`.

- In the old version you would get 1 `Error: Hook disabled` everytime it fires the hook (`watch: true`)
- In the new version, it keeps spamming the error in the console.

Rest worked great! 🙌

File: packages/nextjs/hooks/scaffold-eth/useScaffoldEventHistory.ts (hunk 1 above)
Comment (rin-st) [reply packages/nextjs/hooks/scaffold-eth/useScaffoldEventHistory.ts:49]: I think no, so we can distinguish when there is contract loading `deployedContractLoading`, and when we're reading events `isLoading`

File: packages/nextjs/hooks/scaffold-eth/useScaffoldEventHistory.ts (hunk 2 above)
Comment (rin-st) [reply packages/nextjs/hooks/scaffold-eth/useScaffoldEventHistory.ts:140]: Thank you Pablo, good catch! Fixed [25b4295](https://github.com/scaffold-eth/scaffold-eth-2/pull/827/commits/25b42955b14260514ce7d68d0630559cbf814750). There was a problem that in `catch` block events was set to empty array even it was already empty array and `useEffect` restarted

Comment (Pabl0cks) [reply packages/nextjs/hooks/scaffold-eth/useScaffoldEventHistory.ts:140]: Working great now, TYSM!

**Review by Pabl0cks — APPROVED** [review]
Fix working great! Thanks for the quick fix! 🙌

**Comment by rin-st** [comment]
Thanks! Merging this

**Pull request merged**